Second commit changes

rickonnen opened the pull request:
Hi Carlos here is the exercises about Shapes and the Movies refactoring

I have a question about the final addition to method parameters required by the checkstyle.

rickonnen committed:
Adding the refactor of the Shape example

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Second commit changes

Giovanny Uscamaita

The commit changes include the two exercises about Shapes and Moviews refactoring

Note: I have a doubt about why the checkstyle had asked me to convert most method parameters to final.

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/NewRelease.java:
```diff
@@ -0,0 +1,49 @@
+            return POSITIVE;
+        } else {
+            return NEGATIVE;
+        }
```
instead of having two returns
try this.
```java
return daysRented > 1 ? POSITIVE : NEGATIVE;
```

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/RentalStatement.java:
```diff
@@ -0,0 +1,142 @@
+ */
+public class RentalStatement {
+    private String name;
+    private List<Rental> rentals = new ArrayList<Rental>();
```
this should be instantiate in the constructor

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/RentalStatement.java:
```diff
@@ -0,0 +1,142 @@
+    private String name;
+    private List<Rental> rentals = new ArrayList<Rental>();
+    private double totalAmount;
+    private int frequentRenterPoints;
```
instead of saving as attributes of class
you can have two methods to calculateTotalAmount and calculateTotalFrequentRenterPoints.
if you do that, it will not be necessary to have the clearTotals method. :)

carledriss commented on src/test/java/com/jalasoft/selenium/giovanny/shape/CircleTest.java:
```diff
@@ -0,0 +1,65 @@
+
+        //Then
+        final double expectedArea = 25 * PI;
+        Assert.assertEquals(expectedArea, actualArea, DELTA);
```
Add a static import for Assert.
Please apply in all your tests.

Commit:
Third commit
Applied changes according to your observations.

New method created for total amounts and frequent rent points.

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/shape/Square.java:
```diff
@@ -0,0 +1,43 @@
+ * @author      Giovanny Uscamaita
+ * @version     2.0, 4/24/2017.
+ */
+public class Square implements IShape {
```
Square should extends from Rectangle

carledriss replied:
in order to inherent its methods.

Commit:
Commit for Square changes:

Now the Square class inherits from Rectangle, the constructor inherits from super and also the methods are inherited. All tests pass

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/RentalStatement.java:
```diff
@@ -0,0 +1,139 @@
+     * Method that returns the amount owed by the client.
+     *
+     * @return the value
+     */
```
remove unnecessary comments

carledriss commented on src/test/java/com/jalasoft/selenium/giovanny/movies/MoviesTaskTest.java:
```diff
@@ -0,0 +1,97 @@
+ * @author      Giovanny Uscamaita
+ * @version     2.0, 4/24/2017
+ */
+public class MoviesTaskTest {
```
Try to split this test class
ChildrenTest, NewRealeaseTest, RegularTest

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/RentalStatement.java:
```diff
@@ -0,0 +1,139 @@
+     */
+    private String makeHeader() {
+        String result = "Rental Record for " + getName() + "\n";
+        return result;
```
this could be one line of code
```java
return "Rental Record for " + getName() + "\n";
```

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/RentalStatement.java:
```diff
@@ -0,0 +1,139 @@
+            buf.append(makeRentalLine(rental));
+        }
+        String result =  buf.toString();
+        return result;
```
this could be
```java
return  buf.toString();
```

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/RentalStatement.java:
```diff
@@ -0,0 +1,139 @@
+     */
+    private String formatRentalLine(final Rental rental, final double thisAmount) {
+        String result = "\t" + rental.getTitle() + "\t" + thisAmount + "\n";
+        return result;
```
return directly the result

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/RentalStatement.java:
```diff
@@ -0,0 +1,139 @@
+        String result = "Amount owed is" + totalAmount() + "\n"
+                + "You earned " + totalFrequentRentPoints()
+                + " frequent renter points\n";
+        return result;
```
Same as above

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/Childrens.java:
```diff
@@ -0,0 +1,52 @@
+ *
+ * Created by xianqe on 3/6/2017.
+ */
+public class Childrens extends Movie {
```
Typo

-> Children

carledriss commented on src/main/java/com/jalasoft/selenium/giovanny/movies/Regular.java:
```diff
@@ -0,0 +1,51 @@
+    @Override
+    public double determineAmount(final int daysRented) {
+        double thisAmount = 0;
+        thisAmount += POSITIVE;
```
this could be 
```java
double thisAmount = POSITIVE;
```
